src/bcf_clean_overpar.cpp

```diff
@@ -381,6 +381,17 @@ List bcfoverparRcppClean(NumericVector y_, NumericVector z_, NumericVector w_,
       Rcout << "Current Fit : " <<  logBuff << "\n";
     }
 
+    for (int k=0; k<n; ++k){
+      weight[k] = w[k]*mscale*mscale/(sigma * sigma); // for non-het case, weights need to be divided by sigma square to make it similar to phi
+    }
+
+    for(size_t k=0; k<ntrt; ++k) {
+      weight_het[k] = w[k]*bscale1*bscale1/(sigma*sigma);
+    }
+    for(size_t k=ntrt; k<n; ++k) {
+      weight_het[k] = w[k]*bscale0*bscale0/(sigma*sigma);
+    }
+
```

Comment:
The weights only depend on sigma, w, and bscale, which only change once per MCMC iteration, we can move this loop outside the tree loop for a small gain.